### ISSUE TITLE
Hanging on exiting Java process

### ISSUE THREAD

**Issue by pnathan** [spotify/docker-client#419]
Hi, I have a sample program that's hanging on exit. I have a notion there's not closing properly, but I don't _think_ so

```
package docker;
import com.spotify.docker.client.exceptions.DockerException;
import com.spotify.docker.client.messages.Container;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import com.spotify.docker.client.*;

import java.util.List;

public class Example {
   private static final Logger LOGGER = LoggerFactory.getLogger(MuxDemux.class);
   public static void main(String[] args) throws DockerException, InterruptedException {
      final DockerClient docker = new DefaultDockerClient("unix:///var/run/docker.sock");

      final List<Container> containers = docker.listContainers();

      LOGGER.info("Found " + Integer.toString(containers.size())) ;
      for (int i = 0; i < containers.size(); i++) {

         LOGGER.info("Detected container id: " + containers.get(i).id());
      }


     docker.close();
     LOGGER.info("Going home now") ;
   }
}
```

This code gets the container IDs, prints them, logs "Going home now", and then the program hangs; I have to control-C it.

My expectation is that docker.close() appropriately closes all waiting connections.

Two questions: (1) should this work (i.e., is this PEBKAC (always a possibility. :-) ), and if not, (2) what should I do to fix it?
- docker version: docker 1.10.3 on Debian
- docker-client version: 4.0.2
- Java 1.7

**Comment by plamentotev** [comment]
if it hangs for about a minute then this is the same issue as #405

**Comment by davidxia** [comment]
Should be fixed by #558.
